packages/ember-glimmer/tests/integration/helpers/inline-if-test.js

```diff
@@ -0,0 +1,144 @@
+
+    this.assertText('no');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('no');
+
+    this.runTask(() => set(this.context, 'falsy', 'NO!'));
+
+    this.assertText('NO!');
+
+    this.runTask(() => set(this.context, 'falsy', 'no'));
+
+    this.assertText('no');
+  }
+
+  ['@test it respects isTruthy when object changes']() {
+    this.render(`{{if condition 'truthy' 'falsy'}}`, { condition: EmberObject.create({ isTruthy: false }) });
+
+    this.assertText('falsy');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('falsy');
+
+    this.runTask(() => set(this.context, 'condition', EmberObject.create({ isTruthy: true })));
+
+    this.assertText('truthy');
+
+    this.runTask(() => set(this.context, 'condition', EmberObject.create({ isTruthy: false })));
+
+    this.assertText('falsy');
+  }
+
+  ['@test it respects length test when list itself']() {
+    this.render(`{{if condition 'truthy' 'falsy'}}`, { condition: [] });
+
+    this.assertText('falsy');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('falsy');
+
+    this.runTask(() => set(this.context, 'condition', [1]));
+
+    this.assertText('truthy');
+
+    this.runTask(() => set(this.context, 'condition', []));
+
+    this.assertText('falsy');
+  }
+
+  ['@test it updates when given a falsey second argument']() {
+    this.render(`{{if condition '' 'falsy'}}`, { condition: false });
+
+    this.assertText('falsy');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('falsy');
+
+    this.runTask(() => set(this.context, 'condition', true));
+
+    this.assertText('');
+
+    this.runTask(() => set(this.context, 'condition', false));
+
+    this.assertText('falsy');
+  }
+
+  ['@test it raises when there are more than three arguments']() {
+    expectAssertion(() => {
+      this.render(`{{if condition 'a' 'b' 'c'}}`, { condition: true });
+    }, /The inline form of the `if` and `unless` helpers expect two or three arguments/);
+  }
+
+  ['@test it raises when there are less than two arguments']() {
+    expectAssertion(() => {
+      this.render(`{{if condition}}`, { condition: true });
+    }, /The inline form of the `if` and `unless` helpers expect two or three arguments/);
+  }
+
+}, BASIC_TRUTHY_TESTS, BASIC_FALSY_TESTS);
+
+moduleFor('@glimmer Helpers test: {{if}} used with another helper', class extends SharedHelperConditionalsTest {
+
+  templateFor({ cond, truthy, falsy }) {
+    return `{{concat (if ${cond} ${truthy} ${falsy})}}`;
+  }
+
+});
+
+moduleFor('@glimmer Helpers test: {{if}} used in attribute position', class extends SharedHelperConditionalsTest {
+
+  wrapperFor(templates) {
+    return `<div data-foo="${templates.join('')}" />`;
+  }
+
+  templateFor({ cond, truthy, falsy }) {
+    return `{{if ${cond} ${truthy} ${falsy}}}`;
+  }
+
+  textValue() {
+    return jQuery('div', this.element).attr('data-foo');
+  }
```

Comment:
FYI, I added `this.$ = jQuery;` to `RenderingTest` as `this.$` wasn't already available